backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java

```diff
@@ -2,58 +2,199 @@
-			return null;
-		}
-		Instant isoDateInstant;
-		try {
-      		isoDateInstant = Instant.parse(userSuppliedIsoDateString);
-		}
-		catch (DateTimeParseException parseException) {
-			throw new IllegalGraphqlArgumentException("[" + userSuppliedIsoDateString + "] is not a valid date.");
-		}
-      	return Date.from(isoDateInstant);
-	}
-
-	public static String parsePhoneNumber(String userSuppliedPhoneNumber) {
-		if (userSuppliedPhoneNumber == null) {
-			return null;
-		}
-
-		try {
-			var phoneUtil = PhoneNumberUtil.getInstance();
-			return phoneUtil.format(phoneUtil.parse(userSuppliedPhoneNumber, "US"),
-					PhoneNumberUtil.PhoneNumberFormat.NATIONAL);
-		} catch (NumberParseException parseException) {
-			throw new IllegalGraphqlArgumentException("[" + userSuppliedPhoneNumber + "] is not a valid phone number.");
-		}
-	}
+    private static DateTimeFormatter US_SLASHDATE_SHORT_FORMATTER = DateTimeFormatter.ofPattern("M/d/yyyy");
+
+    public static final LocalDate parseUserShortDate(String d) {
+        String date = parseString(d);
+        if (date == null) {
+            return null;
+        }
+        try {
+            return LocalDate.parse(date, US_SLASHDATE_SHORT_FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new IllegalGraphqlArgumentException("[" + d + "] is not a valid date.");
+        }
+    }
+
+    public static final LocalDate parseUserDate(String userSuppliedDateString) {
+        if (userSuppliedDateString == null) {
+            return null;
+        }
+        if (userSuppliedDateString.contains("/")) {
+            return LocalDate.parse(userSuppliedDateString, US_SLASHDATE_FORMATTER);
+        } else {
+            return LocalDate.parse(userSuppliedDateString); // ISO_LOCAL_DATE is the default
+        }
+    }
+
+    public static final Date parseUserDateTime(String userSuppliedIsoDateString) {
+        if (userSuppliedIsoDateString == null) {
+            return null;
+        }
+        Instant isoDateInstant;
+        try {
+              isoDateInstant = Instant.parse(userSuppliedIsoDateString);
+        }
+        catch (DateTimeParseException parseException) {
+            throw new IllegalGraphqlArgumentException("[" + userSuppliedIsoDateString + "] is not a valid date.");
+        }
+          return Date.from(isoDateInstant);
+    }
+
+    public static String parsePhoneNumber(String userSuppliedPhoneNumber) {
+        if (userSuppliedPhoneNumber == null) {
+            return null;
+        }
+
+        try {
+            var phoneUtil = PhoneNumberUtil.getInstance();
+            return phoneUtil.format(phoneUtil.parse(userSuppliedPhoneNumber, "US"),
+                    PhoneNumberUtil.PhoneNumberFormat.NATIONAL);
+        } catch (NumberParseException parseException) {
+            throw new IllegalGraphqlArgumentException("[" + userSuppliedPhoneNumber + "] is not a valid phone number.");
+        }
+    }
+
+    public static String parseString(String value) {
```

Comment:
I wonder if we should set some large length limitation for strings, like 250 characters, just to be sure nobody goes crazy with one. It doesn't seem like we are likely to have any fields where someone should be writing a novel.

Reply:
Added a limit of 500 characters to be extra safe that no one should hit the limit